rechunker/api.py

```diff
@@ -579,3 +585,45 @@ def _setup_array_rechunk(
     int_proxy = ArrayProxy(int_array, int_chunks)
     write_proxy = ArrayProxy(target_array, write_chunks)
     return CopySpec(read_proxy, int_proxy, write_proxy)
+
+
+@contextlib.contextmanager
+def rechunk_cm(
+    source,
+    target_chunks,
+    max_mem,
+    target_store: str,
+    target_options=None,
+    temp_store: str | None = None,
+    temp_options=None,
+    executor: str | CopySpecExecutor = "dask",
+    filesystem: str | AbstractFileSystem = LocalFileSystem(),
+    keep_target_store: bool = True,
+) -> Rechunked:
+    try:
+        if isinstance(filesystem, str):
+            filesystem = fsspec.filesystem(filesystem)
```

Comment:
This should take the same options as whatever filesystem we are going to operate on (target_options or temp_options?)

Reply:
I am confused here why there are temp_options and target_options, but only one filesystem - that's stopping me from knowing which to use.

Reply:
I'm away for a few days so feel free to take over on this PR!
If there are still some changes needed when I get back, I will try to handle them.

Reply:
There are now 2 filesystem, one for each store.